docs/decisions/20240103-use-jwt-claims-for-agent-admin-auth.md

```diff
@@ -0,0 +1,155 @@
+# Use JWT claims for agent admin access control
+
+- Status: accepted
+- Deciders: Pat Losoponkul, Yurii Shynbuiev, David Poltorak, Shailesh Patil
+- Date: 2024-01-03
+- Tags: multitenancy, authorisation, authentication
+
+Technical Story: [Allow agent admin role to be authenticated by Keycloak JWT | https://input-output.atlassian.net/browse/ATL-6074]
+
+## Context and Problem Statement
+
+External IAM authentication and authorization are currently limited to tenants using the wallets.
+Administrators currently rely on a static API key configured in the agent.
```

Comment:
I would remove the first statement and leave this one as a primary challenge: the prism agent administration is not controlled by a centralized IAM system and is a matter of static API token configuration. This is absolutely ok for the community edition, but doesn't fit to general deployment (aka enterprise edition). The security audit and TARA tickets point us to these problems.

Reply:
done